pkg/bpf/map_linux.go

```diff
@@ -469,17 +475,17 @@ func (m *Map) OpenOrCreate() error {
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
-	return m.openOrCreate(true)
+	return m.openOrCreate(true, true)
 }
 
 // CreateUnpinned creates the map without pinning it to the file system.
 //
 // TODO(tb): Remove this when all map creation takes MapSpec.
-func (m *Map) CreateUnpinned() error {
+func (m *Map) CreateUnpinned(register bool) error {
```

Comment:
Nit: should we add a brief explanation of the `register` parameter in the method docstring?